core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -394,53 +396,68 @@ private long checkDeletedAndGCed(VersionValue versionValue) {
     }
 
     @Override
-    public boolean index(Index index) {
-        final boolean created;
+    public void index(Index index) {
         try (ReleasableLock lock = readLock.acquire()) {
             ensureOpen();
             if (index.origin().isRecovery()) {
                 // Don't throttle recovery operations
-                created = innerIndex(index);
+                innerIndex(index);
             } else {
                 try (Releasable r = throttle.acquireThrottle()) {
-                    created = innerIndex(index);
+                    innerIndex(index);
                 }
             }
-        } catch (IllegalStateException | IOException e) {
-            try {
-                maybeFailEngine("index", e);
-            } catch (Exception inner) {
-                e.addSuppressed(inner);
+        } catch (OutOfMemoryError | IllegalStateException t) {
```

Comment:
I agree this is not inline with our current trend, but it’s a different change imo. Maybe deal with it (and the discussion that will follow) on a different PR?

> On 01 Aug 2016, at 12:19, Jason Tedor <notifications@github.com> wrote:
> 
> In core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java:
> 
> >                  }
> >              }
> > -        } catch (IllegalStateException | IOException e) {
> > -            try {
> > -                maybeFailEngine("index", e);
> > -            } catch (Exception inner) {
> > -                e.addSuppressed(inner);
> > +        } catch (OutOfMemoryError | IllegalStateException t) {
> 
> Yeah, I do not think that we should be catching OutOfMemoryException or try to fail gracefully upon such here. It would be a disaster if during the handling of that exception (i.e., subsequently trying to fail the engine), another exception is thrown, and the original OutOfMemoryException is lost and never bubbles up to cause the VM to tear itself down.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>